src/EditorFeatures/Core/Implementation/Formatting/CodeCleanupService.cs

```diff
@@ -0,0 +1,200 @@
+                IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.FixInlineVariableDeclarations,
+                new[] { IDEDiagnosticIds.InlineDeclarationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnusedVariables,
+                new[] { "CS0168", "CS0219" });
+            dictionary.Add(FeatureOnOffOptions.FixObjectCollectionInitialization,
+                new[] { IDEDiagnosticIds.UseObjectInitializerDiagnosticId, IDEDiagnosticIds.UseCollectionInitializerDiagnosticId });
+            //dictionary.Add(FeatureOnOffOptions.FixLanguageFeatures,
+            //    new[] { IDEDiagnosticIds. });
+            //IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInDeclarationsDiagnosticId,
+            //    IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInMemberAccessDiagnosticId,
+            //    IDEDiagnosticIds.InlineAsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeWithoutNameCheckDiagnosticsId,
+
+            return dictionary;
+        }
+
+        public Document CleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var oldDocument = document;
+            document = RemoveSortUsings(document, cancellationToken);
+            document = ApplyCodeFixes(document, cancellationToken);
+
+            var codeFixChanges = document.GetTextChangesAsync(oldDocument, cancellationToken).WaitAndGetResult(cancellationToken).ToList();
+
+            // we should do apply changes only once. but for now, we just do it twice, for all others and formatting
+            if (codeFixChanges.Count > 0)
+            {
+                ApplyChanges(oldDocument, codeFixChanges, selectionOpt: null, cancellationToken);
+            }
+
+            return document;
+        }
+
+        /// <summary>
+        /// TODO: copied from FormatCommandHandler.cs, will refactor to a better place
+        /// </summary>
+        /// <param name="document"></param>
+        /// <param name="changes"></param>
+        /// <param name="selectionOpt"></param>
+        /// <param name="cancellationToken"></param>
+        private void ApplyChanges(Document document, IList<TextChange> changes, TextSpan? selectionOpt, CancellationToken cancellationToken)
+        {
+            if (selectionOpt.HasValue)
+            {
+                var ruleFactory = document.Project.Solution.Workspace.Services.GetService<IHostDependentFormattingRuleFactoryService>();
+
+                changes = ruleFactory.FilterFormattedChanges(document, selectionOpt.Value, changes).ToList();
+                if (changes.Count == 0)
+                {
+                    return;
+                }
+            }
+
+            using (Logger.LogBlock(FunctionId.Formatting_ApplyResultToBuffer, cancellationToken))
```

Comment:
this should be here. service should never apply to workspace, caller of the service should apply to workspace.

serivce should be document => document tranformation.

Reply:
fixed